src/main/java/macrobase/analysis/pipeline/OneShotPipeline.java

```diff
@@ -0,0 +1,36 @@
+package macrobase.analysis.pipeline;
+
+import macrobase.analysis.result.AnalysisResult;
+import macrobase.conf.ConfigurationException;
+import macrobase.conf.MacroBaseConf;
+
+import java.io.IOException;
+import java.sql.SQLException;
+
+
+abstract public class OneShotPipeline extends AbstractPipeline {
```

Comment:
Ugh, this is ugly. Not a huge fan of returning an Iterator if we only get results only once.